File: Evaluate Reverse Polish Notation - Leetcode 150/Evaluate Reverse Polish Notation - Leetcode 150.java
public class EvaluateReversePolishNotation{
    public static int evalRPN(string[] tokens){
        Stack<Character> stack = new Stack<>();

        for (String t : tokens)
        {
            if ("+-*/".contains(t))
            {
                int b = stack.pop();
                int a = stack.pop();

                switch(t){
                    case "+":
                        stack.push(a+b);
                        break;
                    case "-":
                        stack.push(a-b);
                        break;
                    case "*":
                        stack.push(a*b);
                        break;
                    case "/":
                        int division = a / b;
                        if(division < 0)
                        {
                            stack.push((int) Math.ceil((double) a / b));
                        } else {
                            stack.push((int) Math.floor((double) a / b));
                        }
                        break;
                }
            } else {
                stack.push(Integer.parseInt(t));
            }
        }

        return stack[0];
    }
}

File: Evaluate Reverse Polish Notation - Leetcode 150/Evaluate Reverse Polish Notation - Leetcode 150.ts
// Solution 1: Time - O(n), Space - O(n)
const evalRPN = (tokens: string[]) => {
  const stack: number[] = [];

  for (const t of tokens) {
    if ("+-*/".includes(t)) {
      let a: number, b: number;
      [b, a] = [stack.pop()!, stack.pop()!];

      switch (t) {
        case "+":
          stack.push(a + b);
          break;
        case "-":
          stack.push(a - b);
          break;
        case "*":
          stack.push(a * b);
          break;
        case "/":
          const division = a / b;
          if (division < 0) {
            stack.push(Math.ceil(division));
          } else {
            stack.push(Math.floor(division));
          }
          break;
      }
    } else {
      stack.push(parseInt(t));
    }
  }
  return stack[0];
};
